core/src/test/java/feign/FeignTest.java

```diff
@@ -779,6 +767,51 @@ public void mapAndDecodeExecutesMapFunction() {
     assertEquals(api.post(), "RESPONSE!");
   }
 
+  @Test
+  public void queryMapPropertyPojoWithFullGetterProperty() throws Exception {
+    TestInterface api = new TestInterfaceBuilder().queryMapEndcoder(new BeanQueryMapEncoder())
+        .target("http://localhost:" + server.getPort());
+
+    PropertyPojo.ChildPojoClass propertyPojo = new PropertyPojo.ChildPojoClass();
+    propertyPojo.setAddress("address");
+    propertyPojo.setName("Name");
+    propertyPojo.setNumber(1);
+
+    server.enqueue(new MockResponse());
+    api.queryMapPropertyPojo(propertyPojo);
+    assertThat(server.takeRequest())
+        .hasQueryParams(Arrays.asList("name=Name", "number=1"));
+  }
+
+  @Test
+  public void queryMapPropertyPojoWithPartialGetterProperty() throws Exception {
+    TestInterface api = new TestInterfaceBuilder().queryMapEndcoder(new BeanQueryMapEncoder())
+        .target("http://localhost:" + server.getPort());
+
+    PropertyPojo.ChildPojoClass propertyPojo = new PropertyPojo.ChildPojoClass();
+    propertyPojo.setAddress("address");
+    propertyPojo.setName(null);
+    propertyPojo.setNumber(1);
+
+    server.enqueue(new MockResponse());
+    api.queryMapPropertyPojo(propertyPojo);
+    assertThat(server.takeRequest())
+        .hasQueryParams("number=1");
```

Comment:
Does this test prove that neither "address", nor "name" are in the query params? If so, why isn't "address" there? If not, this test doesn't prove anything that the other tests don't already prove.

Reply:
@nonpool I think this is all we have left

Reply:
@nonpool we are still waiting on your response to this feedback.

Reply:
@kdavisk6 I have changed "address" to "privateGetterProperty" and a test for null value will be ignored another test for testing private getter property will be ignored